src/libraries/Builder/Microsoft.Agents.Builder/App/AgentApplicationOptions.cs

```diff
@@ -9,19 +9,73 @@
 using Microsoft.Agents.Builder.App.UserAuth;
 using Microsoft.Agents.Storage;
 using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace Microsoft.Agents.Builder.App
 {
     /// <summary>
-    /// Options for the <see cref="AgentApplication"/> class.
+    /// TurnState for an Agent is created at the beginning of a Turn by invoking the specified factory delegate.
     /// </summary>
+    /// <remarks>
+    /// The default implementation of this is
+    /// <code>
+    ///    () => new TurnState(service.GetService&lt;IStorage&gt;()
+    /// </code>
+    /// If <c>IStorage</c> was not DI'd, an (effectively) singleton instance should be used:
+    /// <code>
+    ///    var storage = new MemoryStorage();  // MemoryStorage for local dev
+    ///    services.AddTransient&lt;IAgent&gt;(sp =>
+    ///    {
+    ///        var options = new AgentApplicationOptions()
+    ///        {
+    ///           TurnStateFactory = () => new TurnState(storage);
+    ///        };
+    ///        
+    ///        var app = new AgentApplication(options);
+    ///
+    ///        ...
+    ///
+    ///        return app;
+    ///    });
+    /// </code>
+    /// </remarks>
+    /// <seealso cref="TurnState"/>
+    /// See MemoryStorage, BlobsStorage, or CosmosDbStorage.
+    public delegate ITurnState TurnStateFactory();
+
+    /// <summary>
+    /// Options for the <see cref="AgentApplication"/> class.  AgentApplicationOptions can be constructed
+    /// via <c>IConfiguration</c> values or programmatically.
+    /// </summary>
+    /// <seealso cref="TurnStateFactory"/>
+    /// <seealso cref="UserAuthorizationOptions"/>
     public class AgentApplicationOptions
     {
+        /// <summary>
+        /// Constructs AgentApplicationOptions programmatically.
+        /// <code>
+        ///   var options = new AgentApplicationOptions()
```

Comment:
If IStorage is required for TurnState.. can we require it in this ctor?

Reply:
Lets chat about this.  We could.  I hadn't thought through it.  My hesitation is that the AgentApplicationOptionsBuilder doesn't, and relies on a Fluent call to add it.  I'm open though.